### PR TITLE
Add BenchmarkCacheFalseSharing

### PR DESCRIPTION
**Pull request by JamesGuthrie**
## Description

It seems as though the existing benchmarks cannot be used to measure the
effect of false sharing, so I added a new benchmark.

## Merge requirements

Please take into account the following non-code changes that you may need to make with your PR:

- [ ] ~~CHANGELOG entry for user-facing changes~~
- [ ] ~~Updated the relevant documentation~~

### CODE REVIEW

File: pkg/clockcache/cache_bench_test.go [hunk 1]
```diff
@@ -236,6 +237,74 @@ func BenchmarkInsertConcurrent(b *testing.B) {
...
+//
+// Note: This benchmark _must_ be run with the `-cpu=2` argument, to ensure
+// that each goroutine ends up on a different CPU.
+func BenchmarkCacheFalseSharing(b *testing.B) {
+	keys := make([]int, 2)
+	vals := make([]int, 2)
+
+	for i := 0; i < 2; i++ {
+		keys[i], vals[i] = i, i
+	}
```
Comment (arajkumar) [inline pkg/clockcache/cache_bench_test.go:265]: Are these lines are added to avoid compiler optimising this routine?

Comment (JamesGuthrie) [reply pkg/clockcache/cache_bench_test.go]: No, actually this was kinda just "noise" left over from copy-pasting another benchmark. I've taken another pass at this benchmark and simplified it a little.

**Review by arajkumar — APPROVED** [review]
lgtm

File: pkg/clockcache/cache_bench_test.go [hunk 2]
```diff
@@ -236,6 +237,62 @@ func BenchmarkInsertConcurrent(b *testing.B) {
...
+	b.ReportAllocs()
+
+	// define waitgroup, so we can wait until background incrementing tasks are finished
+	wg := &sync.WaitGroup{}
+	wg.Add(2)
+
+	key1 := 0
+	key2 := 1
+	times := b.N
+	b.ResetTimer()
```
Comment (antekresic) [inline pkg/clockcache/cache_bench_test.go:273]: Nit: Could we reset timer in the stressor? That way we ensure the last goroutine to start will reset the timer and just measure after the cache trashing has started.

**Review by antekresic — APPROVED** [review]
Approved with optional suggestion to improve timing the benchmark.

File: pkg/clockcache/cache_bench_test.go (hunk 2 above)
Comment (JamesGuthrie) [reply pkg/clockcache/cache_bench_test.go:273]: I guess we could, but I suspect that could end up distorting the results too. I've done what you suggested, but with an additional waitgroup to coordinate the goroutine start.

**Pull request merged**